src/mbed_tools/project/_internal/libraries.py

```diff
@@ -62,9 +62,12 @@ def fetch(self) -> None:
             git_ref = lib.get_git_reference()
             logger.info(f"Resolving library reference {git_ref.repo_url}.")
             repo = git_utils.clone(git_ref.repo_url, lib.source_code_path)
-            if git_ref.ref:
-                logger.info(f"Checking out revision {git_ref.ref} for library {git_ref.repo_url}.")
-                git_utils.checkout(repo, git_ref.ref)
+
+            if git_ref.ref == "":
+                git_ref.ref = "master"
```

Comment:
Don't hard-code master here. It can easily be wrong. If no ref is given for `fetch()`, clone without doing a checkout. If you are trying to make the local checkout match the latest default branch upstream, you need a different approach.

Reply:
How can we query the "default" branch in a repo?

Reply:
When cloning, no need to query the default branch. The clone will grab the default branch naturally.

When doing a checkout, you may be able to use `git checkout origin/HEAD` to move to the correct git hash, but this does not create a local branch tracking the remote default branch.